pkgs/development/compilers/pforth/default.nix

```diff
@@ -14,14 +15,25 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-vEjFeHSJl+yAtatYJEnu+r9hmOr/kZOgIbSUXR/c8WU=";
   };
 
-  dontConfigure = true;
-
-  preBuild = ''
-    cd platforms/unix
+  # We build the dictionary in a cross-compile compatible way.
+  # For that, we perform steps, that the Makefile would otherwise do.
+  buildPhase = ''
+    runHook preBuild
+    make -C platforms/unix pfdicapp
+    (cd fth/
```

Comment:
```suggestion
    pushd fth/
```

Reply:
I am not too hung up on this either, so I can use pushd and popd as well, but subshells seemed to be a bit clearer. Is there anything wrong with subshells or is this just a styleguide?

Reply:
I updated the code. I just want to say that the subshell has the advantage, that you cannot forget to close it because omitting the `)` is a syntax error, but forgetting the `popd` is not. But it should be good enough for this case.